File: Stack/Valid-Parentheses.js
const pairs = {
    "(": ")",
    "[": "]",
    "{": "}"
}
     
var isValid = function(s) {
    if(s.length % 2 === 1) return false
    if(s[0] === ')' || s[0] === ']' || s[0] === '}') return false
    if(pairs[s[s.length - 1]]) return false
    let stack = []
    for(let i = 0; i < s.length; i++) {
        if(pairs[s[i]]) {
            stack.push(s[i])
            continue
        }
        if(s[i] === pairs[stack[stack.length - 1]]) {
            stack.pop()
        }
        else return false

    }
    if(stack.length) return false
    return true
}

File: Stack/validParentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        if len(s) < 2:
            return False 
       
        stack = []

        for p in s:
            if (p == ')' or p == ']' or p == '}') and len(stack) == 0:
                return False

            if p == '(' or p == '[' or p =='{':
                stack.append(p)
            elif p == ')' and stack[-1] != '(':
                return False
            elif p == ']' and stack[-1] != '[':
                return False
            elif p == '}' and stack[-1] != '{':
                return False
            else:
                stack.pop()
        return len(stack) == 0
